rust-runtime/aws-smithy-observability/src/global.rs

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Utilities for interacting with the currently set `GlobalTelemetryProvider`
+
+use once_cell::sync::Lazy;
+use std::{
+    mem,
+    sync::{Arc, RwLock},
+};
+
+use crate::provider::{GlobalTelemetryProvider, TelemetryProvider};
+
+// Statically store the global provider
+static GLOBAL_TELEMETRY_PROVIDER: Lazy<RwLock<GlobalTelemetryProvider>> =
+    Lazy::new(|| RwLock::new(GlobalTelemetryProvider::new(TelemetryProvider::default())));
+
+/// Set the current global [TelemetryProvider]. If [None] is supplied then a noop provider is set.
+/// The previous [TelemetryProvider] is returned in an [Arc] so appropriate cleanup can be done if necessary.
+pub fn set_global_telemetry_provider(
+    new_provider: Option<TelemetryProvider>,
```

Comment:
If `TelemetryProvider::default()` is no-op or we offer `TelemetryProvider::noop()` (which we probably should anyway as it will be less ambiguous) then we don't need `Option<T>` here just take in the telemetry provider directly.

Reply:
Added an explicit `noop` method for `TelemetryProvider` in https://github.com/smithy-lang/smithy-rs/pull/3912/commits/0559e8b54fb41bcf04125e2e3cedeec87b24a9f9